tools/base/utils.py

```diff
@@ -9,10 +9,16 @@
 import tempfile
 from configparser import ConfigParser
 from contextlib import ExitStack, contextmanager, redirect_stderr, redirect_stdout
-from typing import Callable, Iterator, List, Optional, Union
+from typing import Callable, Iterator, List, Optional, Set, Union
 
 import yaml
 
+# See here for a list of known tar file extensions:
+#   https://en.wikipedia.org/wiki/Tar_(computing)#Suffixes_for_compressed_files
+# not all are listed here, and some extensions may require additional software
+# to handle. This list can be updated as required
+TAR_EXTS: Set[str] = {"tar", "tar.gz", "xz", "tar.xz", "tar.bz2"}
```

Comment:
Why .xz, which is compressed, but doesn't imply an archive? Definitely add .tgz as a well known compound extension.